internal/integration/sessions_test.go

```diff
@@ -584,6 +583,81 @@ func TestSessionsProse(t *testing.T) {
 	})
 }
 
+func TestSessionsProse_21_SettingSnapshotTimeWithoutSnapshot(t *testing.T) {
+	// 21. Having snapshotTime set and snapshot set to false is not allowed.
+	mtOpts := mtest.
+		NewOptions().
+		MinServerVersion("5.0").
+		Topologies(mtest.ReplicaSet, mtest.Sharded)
+
+	mt := mtest.New(t, mtOpts)
+
+	mt.Setup()
```

Comment:
Optional: Consider combining these two calls into a new function to make it easier to use, like `mtest.Setup`. We can reconcile the two entry points in [GODRIVER-3656](https://jira.mongodb.org/browse/GODRIVER-3656).

E.g.
```
func TestBlah(...) {
	mt := mtest.Setup(t, mtOpts)
	// ...
}
```
```
package mtest

func Setup(...) *T {
	mt := mtest.New(...)
	mt.setup()
	// ...
	return mt
}
```